src/wolff_producers.erl

```diff
@@ -529,3 +540,31 @@ ensure_timer_cancelled(Tref) when is_reference(Tref) ->
   ok;
 ensure_timer_cancelled(_) ->
   ok.
+
+supervisor_child_id(St) ->
```

Comment:
Should the supervisor use this same function to make it consistent by construction?

Reply:
producer id is not exactly the supervisor child id.
i'll make it simpler